### PR TITLE
enh(filters): use select 2 for the poller filter

### PR DESCRIPTION
**Pull request by kduret**
## Description

allows people to use the almighty select 2 instead of a text field. 

Blocked by https://github.com/centreon/centreon/pull/11660

image

image

**Fixes** MON-10870

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

### CODE REVIEW

File: engine-status/index.php [hunk 1]
```diff
@@ -93,75 +93,100 @@
...
-$res->execute();
-while ($row = $res->fetch()) {
-    $dataSth[] = $row;
-}
+    while ($row = $res2->fetchRow()) {
+      $row['h_max'] = round($row['h_max'], 3);
+      $row['h_moy'] = round($row['h_moy'], 3);
+      $row['s_max'] = round($row['s_max'], 3);
+      $row['s_moy'] = round($row['s_moy'], 3);
+      $dataEx[] = $row;
```
Comment (kduret) [inline engine-status/index.php:148]:
```suggestion
        $row['h_max'] = round($row['h_max'], 3);
        $row['h_moy'] = round($row['h_moy'], 3);
        $row['s_max'] = round($row['s_max'], 3);
        $row['s_moy'] = round($row['s_moy'], 3);
        $dataEx[] = $row;
```

File: engine-status/index.php [hunk 2]
```diff
@@ -93,75 +93,100 @@
...
-    $row['s_max'] = round($row['s_max'], 3);
-    $row['s_moy'] = round($row['s_moy'], 3);
-    $dataEx[] = $row;
-}
+if (!empty($queryPoller)) {
+    $queryLat = "SELECT MAX(T1.latency) AS h_max, AVG(T1.latency) AS h_moy,
+            MAX(T2.latency) AS s_max, AVG(T2.latency) AS s_moy
+            FROM hosts T1, services T2
+            WHERE T1.instance_id IN (" . $queryPoller . ")
+            AND T1.host_id = T2.host_id AND T2.enabled = '1';";
```
Comment (jeremyjaouen) [inline engine-status/index.php:119]:
```suggestion
            AND T1.host_id = T2.host_id AND T2.enabled = '1'";
```

File: engine-status/index.php [hunk 3]
```diff
@@ -93,75 +93,100 @@
...
+    $queryEx = "SELECT MAX(T1.execution_time) AS h_max, AVG(T1.execution_time) AS h_moy,
+            MAX(T2.execution_time) AS s_max, AVG(T2.execution_time) AS s_moy
+            FROM hosts T1, services T2
+            WHERE T1.instance_id IN (" . $queryPoller . ") AND T1.host_id = T2.host_id
+            AND T2.enabled = '1';";
```
Comment (jeremyjaouen) [inline engine-status/index.php:124]:
```suggestion
            AND T2.enabled = '1'";
```

File: engine-status/index.php [hunk 4]
```diff
@@ -93,75 +93,100 @@
...
-    $dataSts[] = $row;
+    $querySth = "SELECT SUM(CASE WHEN h.state = 1 AND h.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Dow,
+                SUM(CASE WHEN h.state = 2 AND h.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Un,
+                SUM(CASE WHEN h.state = 0 AND h.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Up,
+                SUM(CASE WHEN h.state = 4 AND h.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Pend
+                FROM hosts h WHERE h.instance_id IN (" . $queryPoller . ");";
```
Comment (jeremyjaouen) [inline engine-status/index.php:159]:
```suggestion
                FROM hosts h WHERE h.instance_id IN (" . $queryPoller . ")";
```

File: engine-status/index.php [hunk 5]
```diff
@@ -93,75 +93,100 @@
...
+                SUM(CASE WHEN s.state = 1 AND s.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Wa,
+                SUM(CASE WHEN s.state = 0 AND s.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Ok,
+                SUM(CASE WHEN s.state = 4 AND s.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Pend,
+                SUM(CASE WHEN s.state = 3 AND s.enabled = 1 AND h.name NOT LIKE '%Module%'
+                THEN 1 ELSE 0 END) AS Unk
+                FROM services s, hosts h
+                WHERE h.host_id = s.host_id AND h.instance_id IN (" . $queryPoller . ");";
```
Comment (jeremyjaouen) [inline engine-status/index.php:172]:
```suggestion
                WHERE h.host_id = s.host_id AND h.instance_id IN (" . $queryPoller . ")";
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by adr-mo — APPROVED** [review]

**Comment by kduret** [comment]
@tanguyvda do you know why `and T2.check_type = '0'` condition has been removed ?

**Comment by tanguyvda** [comment]
I shouldn't have removed that. 
Since the query retrieves latency and execution time, having passive services in it is not useful apart from very specific cases it will just mess up the statistics that we are computing.

File: engine-status/index.php [hunk 6]
```diff
@@ -93,75 +93,103 @@
...
+        if ($queryPoller != '') {
+            $queryPoller .= ', ';
+        }
+        $queryPoller .= ":id_" . $result;
+        $mainQueryParameters[] = [
+            'parameter' => ':id_' . $result,
+            'value' => (int)$result,
+            'type' => PDO::PARAM_INT
+        ];
+    }
```
Comment (TamazC) [inline engine-status/index.php:111]: you could create a array with instance id as key and value like so :
`foreach ($results as $result) {
        $instanceId = ':id_' . $result;
        $instances[$instanceId] = (int) $result;`
        
And than in query something like `WHERE T1.instance_id IN (" . join(', ', array_keys($instances)) . ")`

**Review by TamazC — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by TamazC — APPROVED** [review]

**Pull request merged**